packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -66,12 +68,55 @@ export default class OperatorModeStateService extends Service {
     let itemIndex = this.state.stacks[stackIndex].items.indexOf(item);
 
     if (newItem.stackIndex !== stackIndex) {
-      this.removeItemFromStack(item);
+      this.removeItemFromStack(item, { keepIfEmpty: true });
       this.addItemToStack(newItem);
```

Comment:
Its so important to have an option to differentiate keepIfEmpty when removing an item

For example,
- stack0: item1
- stack1: item2
- want to add item3

If we remove item1, stack1-> stack0. adding item3 will need to refer to index 0 not index 1. Here I add an option keepIfEmpty which prevents the stack from being delete